src/main/java/com/backbase/ct/bbfuel/configurator/ProductSummaryConfigurator.java

```diff
@@ -28,12 +29,31 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ProductSummaryConfigurator.class);
     private final ArrangementsIntegrationRestClient arrangementsIntegrationRestClient;
 
+    private final ArrangementsReader reader = new ArrangementsReader();
+
     public void ingestProducts() {
         List<ProductsPostRequestBody> products = ProductSummaryDataGenerator.getProductsFromFile();
         products.stream().parallel()
             .forEach(arrangementsIntegrationRestClient::ingestProductAndLogResponse);
     }
 
+    public List<ArrangementId> ingestRetailArrangements(String externalLegalEntityId) {
+        List<ArrangementsPostRequestBody> arrangements = synchronizedList(new ArrayList<>());
+        List<ArrangementId> arrangementIds = synchronizedList(new ArrayList<>());
+
+        arrangements.addAll(reader.load(externalLegalEntityId));
```

Comment:
side note: although it is a small file it is not efficient to load it for each retail customer.